In [5]:
import numpy as np
from scipy import stats
from LinearRegression import LinearRegression

data = np.genfromtxt(
    "../Data/housing.csv",
    delimiter=",",
    names=True,
    dtype=None,
    encoding="utf-8")


data.shape, data.dtype.fields

((20640,),
 mappingproxy({'longitude': (dtype('float64'), 0),
               'latitude': (dtype('float64'), 8),
               'housing_median_age': (dtype('float64'), 16),
               'total_rooms': (dtype('float64'), 24),
               'total_bedrooms': (dtype('float64'), 32),
               'population': (dtype('float64'), 40),
               'households': (dtype('float64'), 48),
               'median_income': (dtype('float64'), 56),
               'median_house_value': (dtype('float64'), 64),
               'ocean_proximity': (dtype('<U10'), 72)}))

Vi har n = 20640 observationer. 10 stycken kolonner. 
### 8 numeriska X:longitude,latitude,housing_median_age,total_rooms,total_bedrooms,population,households,median_income,median_house_value
### 1 kategoriskt X: ocean_proximity
### 1 Y: median_house_value


#### Ocean har 5 olika kategorier:
<1H OCEAN - 9136
INLAND - 6551
ISLAND - 5
NEAR BAY - 2290
NEAR OCEAN - 2658

Värt att notera att "ISLAND" är endast 5.

In [ ]:

X = np.column_stack([
    np.ones(len(median_income)),  # intercept
    housing_median_age,
    total_rooms,
    total_bedrooms,
    population,
    households,
    median_income
])

model = LinearRegression()
model.fit(X,y)

corr = model.pearson()
print(corr)

NameError: name 'n' is not defined